feature/experimental/system/health/tests/system_generic_health_check/system_generic_health_check_test.go

```diff
@@ -0,0 +1,745 @@
+	for _, component := range cardList {
+		query := gnmi.OC().Component(component).State()
+		timestamp := time.Now().Round(time.Second)
+		componentState := gnmi.Get(t, dut, query)
+		componentType := componentState.GetType()
+		memoryState := componentState.GetMemory()
+		if memoryState == nil {
+			t.Errorf("ERROR: %s - Device: %s - %s: %-40s - Type: %-20s - Memory data not available",
+				timestamp, deviceName, description, component, componentType)
+			continue
+		}
+
+		memoryAvailable := memoryState.GetAvailable()
+		memoryUtilized := memoryState.GetUtilized()
+		memoryUtilization := uint8((memoryUtilized * 100) / (memoryAvailable + memoryUtilized))
+		if memoryUtilization > memoryHighUtilization {
+			t.Errorf("ERROR: %s - Device: %s - %s: %-40s - Type: %-20s - Utilization: %3d%% (Threshold: %3d%%)",
+				timestamp, deviceName, description, component, componentType, memoryUtilization, memoryHighUtilization)
+		} else {
+			t.Logf("INFO: %s - Device: %s - %s: %-40s - Type: %-20s - Utilization: %3d%%",
+				timestamp, deviceName, description, component, componentType, memoryUtilization)
+		}
+	}
+}
+
+func TestSystemProcessNoHighCPUSpike(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	deviceName := dut.Device.Name()
+
+	query := gnmi.OC().System().ProcessAny().State()
+	description := "System CPU Process"
+	threshold := uint8(cpuHighUtilization)
+
+	timestamp := time.Now().Round(time.Second)
+	results := gnmi.GetAll(t, dut, query)
+	for _, result := range results {
+		processName := result.GetName()
+		utilization := result.GetCpuUtilization()
+
+		if utilization > threshold {
+			t.Errorf("%s %s ERROR: %s process %-40s utilization: %3d%%, exceeding %3d%%", timestamp, deviceName, description, processName, utilization, threshold)
+		} else {
+			t.Logf("%s %s INFO: %s process %-40s utilization: %3d%%", timestamp, deviceName, description, processName, utilization)
+		}
+	}
+}
+
+func TestSystemProcessNoHighMemorySpike(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	deviceName := dut.Device.Name()
+
+	query := gnmi.OC().System().ProcessAny().State()
+	description := "System Process Memory"
+
+	currentTime := time.Now().Round(time.Second)
+	processes := gnmi.GetAll(t, dut, query)
+	for _, process := range processes {
+		processName := process.GetName()
+
+		if process.MemoryUtilization != nil {
+			utilization := process.GetMemoryUtilization()
+			if utilization > memoryHighUtilization {
+				t.Errorf("%s %s ERROR: %s - Process: %-40s - Utilization: %3d%%, exceeding %3d%%", currentTime, deviceName, description, processName, utilization, memoryHighUtilization)
+			} else {
+				t.Logf("%s %s INFO:  %s - Process: %-40s - Utilization: %3d%%", currentTime, deviceName, description, processName, utilization)
+			}
+		} else {
+			t.Logf("%s %s INFO:  %s - Process: %-40s - Utilization data not available", currentTime, deviceName, description, processName)
+		}
+	}
+}
+
+func TestGetInterfaceName(t *testing.T) {
```

Comment:
What is this testing? It's only exercising Ondatra itself, which doesn't seem useful/necessary?

Reply:
TestGetInterfaceName is used to get all the interfaces' name and assign them to a global variable interface. The testing needs to use all the interfaces' name in multiple test cases. Thus, we don't need lookup it every time.